File: android/app/src/main/java/com/mwiacek/przepisy/drogowe/ScrollerWebView.java
package com.mwiacek.przepisy.drogowe;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.webkit.WebView;

//Shared between Straz and Przepisy
//1.0
public class ScrollerWebView extends WebView {
    MotionEvent ev;
    ScrollerView sv;

    public ScrollerWebView(Context context) {
        super(context);
    }

    public ScrollerWebView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ScrollerWebView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        if (event.getAction() == MotionEvent.ACTION_DOWN) ev = event;
        return super.onTouchEvent(event);
    }

    @Override
    protected void onScrollChanged(int x, int y, int oldx, int oldy) {
        super.onScrollChanged(x, y, oldx, oldy);
        if (sv != null && GetMax() != 0) {
            sv.y = (this.getHeight() - sv.myBitmap.getHeight())
                    * computeVerticalScrollOffset() / GetMax();
            sv.SetTimer();
        }
    }

    int GetMax() {
        return computeVerticalScrollRange() - computeVerticalScrollExtent();
    }
}

File: android_kotlin/app/src/main/java/com/mwiacek/przepisy/drogowe/ScrollerWebView.kt
package com.mwiacek.przepisy.drogowe

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.webkit.WebView

//Shared between Straz and Przepisy
//1.0
class ScrollerWebView : WebView {
    var ev: MotionEvent? = null
    var sv: ScrollerView? = null

    constructor(context: Context?) : super(context!!)
    constructor(context: Context?, attrs: AttributeSet?) : super(
        context!!, attrs
    )

    constructor(context: Context?, attrs: AttributeSet?, defStyle: Int) : super(
        context!!, attrs, defStyle
    )

    override fun onTouchEvent(event: MotionEvent): Boolean {
        if (event.action == MotionEvent.ACTION_DOWN) ev = event
        return super.onTouchEvent(event)
    }

    override fun onScrollChanged(x: Int, y: Int, oldx: Int, oldy: Int) {
        super.onScrollChanged(x, y, oldx, oldy)
        if (sv != null && GetMax() != 0) {
            sv!!.y = this.height - sv!!.myBitmap.height * computeVerticalScrollOffset() / GetMax()
            sv!!.SetTimer()
        }
    }

    fun GetMax(): Int {
        return computeVerticalScrollRange() - computeVerticalScrollExtent()
    }
}